### PR TITLE
Everything is a stream

### PR DESCRIPTION
**Pull request by dbrattli**
Streams in, streams out. Now everything is a stream. I've added some more functions `Stream.choose` and `Stream.chooseNamed` to dig into stream you want to send to a component. You still use `Stream.map` to get back out. Looks cleaner, so I hope we are not over-engineering things.

We now need to give the name of the message stream to `withMsgStream` so it knows the default name. Since the name is of type `'name` we cannot give a default ourselves. I think this is OK since it removes many `msgs |> AsyncRx.asStream "msgs"` within the code and it's good not to lock the type to a string.

### CODE REVIEW

File: paket.lock [hunk 1]
```diff
@@ -1202,6 +1196,13 @@ NUGET
     Thoth.Json (2.5) - restriction: >= netstandard2.0
       Fable.Core (>= 2.0) - restriction: >= netstandard2.0
       FSharp.Core (>= 4.5.2) - restriction: >= netstandard2.0
+  remote: /Users/dbrattli/Developer/packages
```
Comment (rommsen) [inline paket.lock]: In think that this was not intended 😄

File: src/Client/Client.fs [hunk 2]
```diff
@@ -120,44 +120,45 @@ let asMagicMsg msg =
...
-let stream (model: Model) (msgs: IAsyncObservable<Msg>) =
+let stream model msgs =
   match model with
   | Loading ->
-      Streams
-        [
-          msgs |> AsyncRx.asStream "msgs"
-          loadLetterString () |> AsyncRx.asStream "loading"
-        ]
+      loadLetterString ()
```
Comment (rommsen) [inline src/Client/Client.fs]: lets inline this now

**Comment by rommsen** [comment]
@dbrattli I think the API looks pretty clean now. Need some time this week to understand what you changed in the backend and to test it. Can you fix the `remote: /Users/dbrattli/Developer/packages` part in paket.lock than I will merge the PR

File: paket.lock (hunk 1 above)
Comment (dbrattli) [reply paket.lock]: Fixed

**Pull request merged**